packages/react-sdk/docusaurus/docs/React/02-tutorials/02-audio-room.mdx

```diff
@@ -3,527 +3,961 @@ title: Audio Room Tutorial
-
-To do this we need to query the backend for a list of the calls and display each of the rooms. Again, we'll not focus on the UI
-part since that's different for each use case.
+One of the main screens of the application will be showing a list of the rooms that are available for users to join. We will query the data using `videoClient.queryCalls()`. We want to group the calls into 3 categories:
 
-So, let's assume we have a component that stores the rooms in a `rooms` array, so let's define that at the root of the component:
+1. **upcoming**
+2. **live**
+3. **ended**
 
-```
-const [rooms, setRooms] = useState<AudioRoom[]>([]);
+```ts
+export type RoomLiveState = 'live' | 'upcoming' | 'ended';
 ```
 
-Next, querying is fairly easy. Here is the code and we'll go through it afterwards:
+The grouping will be performed on the back-end side thanks to the [use of `filter_conditions`](../../guides/querying-calls) in our calls queries. The calls will be queried inside the `RoomListing` component. Note we are implementing pagination with `next` query parameter:
 
-```tsx
-const client = useStreamVideoClient();
-useEffect(() => {
-  client
-    ?.queryCalls({
-      filter_conditions: { audioRoomCall: true },
-      sort: [],
-      watch: true,
-    })
-    .then((result) => {
-      console.log('Querying calls successful.');
-      setRooms(result.calls);
-    })
-    .catch((err) => {
-      console.log('Querying calls failed.');
-      console.error(err);
+```ts
+import { useCallback, useEffect, useRef, useState } from 'react';
+// omitted imports ...
+import {
+  Call,
+  // omitted imports ...
+  QueryCallsRequest,
+  // omitted imports ...
+  useStreamVideoClient
+} from '@stream-io/video-react-sdk';
+import { CALL_TYPE } from '../../contexts';
+// omitted imports
+import { RoomLiveState } from '../../utils/roomLiveState';
+
+const QUERY_CALLS_PARAMS: QueryCallsRequest = {
+  // sort in descending order by `created_at` field
+  sort: [{ direction: -1, field: 'created_at' }],
+  // max 10 items per query
+  limit: 10,
+  // subscribe to receive WS events for every call object queried
+  watch: true,
+};
+
+const CALL_TYPE_FILTER = { type: CALL_TYPE };
+
+const BY_ROOM_STATE_FILTER: Record<
+  RoomLiveState,
+  QueryCallsRequest['filter_conditions']
+> = {
+  upcoming: {
+    ...CALL_TYPE_FILTER,
+    backstage: true,
+    ended_at: null,
+  },
+  live: {
+    ...CALL_TYPE_FILTER,
+    backstage: false,
+    ended_at: null,
+  },
+  ended: {
+    ...CALL_TYPE_FILTER,
+    ended_at: { $lte: new Date().toISOString() },
+  },
+};
+
+export const RoomListing = ({ liveState }: { liveState: RoomLiveState }) => {
+  const client = useStreamVideoClient();
+  const [calls, setCalls] = useState<Call[]>([]);
+  // omitted code initiating state variables ...
+  const nextPage = useRef<string | undefined>();
+
+  const loadCalls = useCallback(async () => {
+    if (!client) return;
+    const result = await client.queryCalls({
+      ...QUERY_CALLS_PARAMS,
```

Comment:
I think we should inline the parameters here. IMO, having these constants reduces the readability of the tutorial, as the reader (me) has to jump back and forth on the page.

My experience:
- I scan the QUERY_CALLS_PARAMS value, but I'm not paying too much attention to it as I lack the context and I proceed forward
- a couple of lines below, I see it being used, and now I have to scroll back and actually see what it really means when used in `queryCalls` context...

Reply:
@oliverlaz I agree. I have inlined the code.